opinions/spring_boot_opinions.go

```diff
@@ -101,129 +116,76 @@ var SpringBoot = Opinions{
 			return bootMetadata.HasDependency("spring-boot-actuator")
 		},
 		ApplyFunc: func(ctx context.Context, podSpec *corev1.PodTemplateSpec, containerIdx int, imageMetadata cnb.BuildMetadata) error {
-			applicationProperties := SpringApplicationProperties(ctx)
-
-			// TODO check for an existing value before clobbering
-			applicationProperties["management.server.port"] = strconv.Itoa(9001)
-			applicationProperties["management.server.ssl.enabled"] = "false"
-			applicationProperties["management.endpoint.health.enabled"] = "true"
-			applicationProperties["management.endpoint.info.enabled"] = "true"
-			applicationProperties["management.endpoints.web.base-path"] = "/actuator"
-
-			podSpec.Annotations["boot.spring.io/actuator"] = fmt.Sprintf("http://:%s%s",
-				applicationProperties["management.server.port"],
-				applicationProperties["management.endpoints.web.base-path"],
-			)
+			applicationProperties := GetSpringApplicationProperties(ctx)
 
-			return nil
-		},
-	},
-	&BasicOpinion{
-		Id: "spring-boot-actuator-probes",
-		ApplicableFunc: func(applied AppliedOpinions, imageMetadata cnb.BuildMetadata) bool {
-			return applied.Has("spring-boot-actuator")
-		},
-		ApplyFunc: func(ctx context.Context, podSpec *corev1.PodTemplateSpec, containerIdx int, imageMetadata cnb.BuildMetadata) error {
-			applicationProperties := SpringApplicationProperties(ctx)
-
-			managementPort, err := strconv.Atoi(applicationProperties["management.server.port"])
-			if err != nil {
-				return err
+			managementPort := applicationProperties.Default("management.server.port", applicationProperties["server.port"])
+			managementBasePath := applicationProperties.Default("management.endpoints.web.base-path", "/actuator")
+			managementScheme := corev1.URISchemeHTTP
+			if applicationProperties["management.server.ssl.enabled"] == "true" {
```

Comment:
Disabling SSL termination in Boot is probably a sane opinion as usually it would be terminated at the cloud providers load balancer or ingress

Reply:
we shouldn't overwrite the user's intent. If they want SSL, they should get it. This check is only detecting if ssl was requested

Reply:
agreed